### PR TITLE
fix: use field labels, rather than names, in FieldMapping dropdowns

### PR DESCRIPTION
**Pull request by khennes**
When using field names, custom fields are displayed with a `__c` suffix, e.g. `FavoriteFruit__c`.

### CODE REVIEW

**Review by albertyfwu — COMMENTED** [review]
Is it possible in Salesforce to create a custom field with `Email` label and `Email__c` name even though an existing property `Email` already exists?

**Comment by khennes** [comment]
> Is it possible in Salesforce to create a custom field with `Email` label and `Email__c` name even though an existing property `Email` already exists?
> 
> Does salesforce enforce uniqueness on field labels in addition to field names?

Apparently not:

Screenshot 2023-02-09 at 10 28 53 AM

**Comment by albertyfwu** [comment]
> > Is it possible in Salesforce to create a custom field with `Email` label and `Email__c` name even though an existing property `Email` already exists?
> > Does salesforce enforce uniqueness on field labels in addition to field names?
> 
> Apparently not:
> 
> Screenshot 2023-02-09 at 10 28 53 AM

given that non-enforcement, imo it might be good to see the `__c` suffix if it's a commonly-understood format by customers using Supaglue. anybody else have thoughts on this?

**Comment by khennes** [comment]
> imo it might be good to see the __c suffix if it's a commonly-understood format by customers

I think the expected behavior would be to see the display label rather than the name. If in the future we see that customers are actually creating identically labeled fields in their Salesforce instances, we can add some kind of text or indicator in the dropdown to disambiguate.

File: packages/nextjs/src/components/FieldMapping/FieldMapping.tsx [hunk 1]
```diff
@@ -142,34 +147,31 @@ const FieldCollection = ({ appearance, syncConfig, sync }: FieldCollectionProps)
...
-        );
-      })}
+      {syncConfig.destination.schema.fields.map(({ label, name }, idx) => (
+        <div
+          key={idx}
+          css={styles.fieldWrapper}
+          className={classNames(appearance?.elements?.fieldWrapper, 'sg-fieldWrapper')}
+        >
+          <p css={styles.fieldName} className={classNames('sg-fieldName', appearance?.elements?.fieldName)}>
+            {label || name}
```
Comment (albertyfwu) [inline packages/nextjs/src/components/FieldMapping/FieldMapping.tsx:157]: nit: `label ?? name`?

**Review by albertyfwu — APPROVED** [review]

File: packages/nextjs/src/components/FieldMapping/FieldMapping.tsx (hunk 1 above)
Comment (khennes) [reply packages/nextjs/src/components/FieldMapping/FieldMapping.tsx:157]: thanks will do!

**Review by albertyfwu — APPROVED** [review]

**Pull request merged**